pkg/resources/reconciling/compare.go

```diff
@@ -38,8 +40,17 @@ func init() {
 
 // DeepEqual compares both objects for equality
 func DeepEqual(a, b metav1.Object) bool {
-	if reflect.DeepEqual(a, b) {
-		return true
+	// The ClusterTemplate consist with ClusterSpec which has nested Semver type.
+	// The semverlib.Version doesn't implement Equality function. Using equality.Semantic.DeepEqual for this object cause
+	// an informative panic.
+	if _, ok := a.(*kubermaticv1.ClusterTemplate); ok {
```

Comment:
BTW a more elegant solution would be to create our `Equalities` instance providing a comparison function for `Semver` type. See the following for reference:

https://github.com/kubernetes/apimachinery/blob/b63a0c883fbfc313249150449400788e5589ef23/pkg/api/equality/semantic.go#L29-L49

Reply:
I will try, thanks

Reply:
I tested it and you can add this to the init function of `compare.go`
Or we can add this to the Add method of the ClusterTemplate controller, but then it only works for the controller, and in future maybe something else will need it. 

```
func init() {
	// Kubernetes Objects can be deeper than the default 10 levels.
	deep.MaxDepth = 20
	deep.LogErrors = true

	// Add comparison function for semver.Version
	equality.Semantic.AddFunc(func(a, b *semver.Version) bool {
		return a.Compare(b) == 0
	})
}

```

Reply:
I had to add other objects to make it run our controller, seems to work fine
Thanks, @irozzo-1A